cli-plugins/plugin/plugin.go

```diff
@@ -24,12 +32,45 @@ import (
 // called.
 var PersistentPreRunE func(*cobra.Command, []string) error
 
+// closeOnParentSocketClose connects to the socket specified
+// by the CLI_PLUGIN_EXIT_SOCKET env var, if present, and attempts
+// to read from it until it receives an EOF, which signals that
+// the CLI is going to exit and the plugin should also exit.
+func closeOnParentSocketClose() {
```

Comment:
Would it be difficult to (could be in a follow-up) write an integration test for this? I recall we already have some tests that build a test-plugin; perhaps we should have a test with a plugin that doesn't terminate immediately (e.g.)

Reply:
I think we should create a follow up to this to investigate how to best do that. I'd like to do it, but I don't think it should be a blocker since this change ultimately falls back to the previous behaviour if it fails.

Reply:
Yup, also fine for after this PR. I was curious if you already had tried (and if you already had ideas for testing)

Reply:
opened https://github.com/docker/cli/issues/4692 as placeholder for that